bitcoin/src/lib.rs

```diff
@@ -59,7 +59,6 @@ compile_error!(
 #[cfg(bench)]
 extern crate test;
 
-#[cfg(feature = "no-std")]
```

Comment:
I don't fully understand the cost/benefits of always using the `alloc` crate vs using std for std builds and alloc only for no-std builds. Not important for this PR but its something I've noticed I didn't understand this week, if you have time ever can you explain your thoughts please?

Reply:
Oh, needs more context, the lines below this are 
```rust
#[macro_use]
extern crate alloc;
```

Reply:
> I don't fully understand the cost/benefits of always using the alloc crate vs using std for std builds and alloc only for no-std builds. 

Just simpler code, less compilation conditions to worry about.